src/watcherEx.ts

```diff
@@ -0,0 +1,227 @@
+    subConnection: RedisConnection | RedisClusterConnection,
+    channel?: string,
+    callBackOptions?: WatcherExCallBackOptions
+  ): Promise<RedisWatcherEx> {
+    const watcherEx = new RedisWatcherEx();
+
+    watcherEx.pubConnection = pubConnection;
+    watcherEx.subConnection = subConnection;
+
+    if (channel) {
+      watcherEx.channel = channel;
+    }
+
+    watcherEx.watcherExCallBackOptions = callBackOptions;
+    const watcherExCallBackMethods: any = {
+      updateForAddPolicy: callBackOptions?.addPolicyCb,
+      updateForRemovePolicy: callBackOptions?.removePolicyCb,
+      updateForRemoveFilteredPolicy: callBackOptions?.removeFilteredPolicyCb,
+      updateForSavePolicy: callBackOptions?.savePolicyCb,
+      updateForAddPolicies: callBackOptions?.addPoliciesCb,
+      updateForRemovePolicies: callBackOptions?.removePoliciesCb
+    };
+
+    const client = await watcherEx.subConnection.getRedisClient();
+    // @ts-ignore - subscribe is exists.
+    await client.subscribe(watcherEx.channel);
+    client.on('message', (chan: string, msg: any) => {
+      if (chan !== watcherEx.channel) {
+        return;
+      }
+      msg = JSON.parse(JSON.stringify(msg));
+      if (watcherExCallBackMethods[msg.updateType]) {
+        watcherExCallBackMethods[msg.updateType](msg.args);
+      }
+    });
+
+    return watcherEx;
+  }
+
+  private constructor() {}
+
+  public async updateForAddPolicy(sec: string, ptype: string, ...params: string[]): Promise<void> {
+    const client = await this.pubConnection.getRedisClient();
+    const message = {
+      updateType: updateTypes.addPolicy,
+      args: {
+        sec,
+        ptype,
+        params
+      }
+    };
+    // @ts-ignore - publish is exists.
+    await client.publish(this.channel, JSON.stringify(message));
+  }
+
+  public async updateForRemovePolicy(sec: string, ptype: string, ...params: string[]): Promise<void> {
+    const client = await this.pubConnection.getRedisClient();
+    const message = {
+      updateType: updateTypes.removePolicy,
+      args: {
+        sec,
+        ptype,
+        params
+      }
+    };
+    // @ts-ignore - publish is exists.
+    await client.publish(this.channel, JSON.stringify(message));
+  }
+
+  public async updateForRemoveFilteredPolicy(sec: string, ptype: string, field: number, ...fieldValues: string[]): Promise<void> {
+    const client = await this.pubConnection.getRedisClient();
+    const message = {
+      updateType: updateTypes.removeFilteredPolicy,
+      args: {
+        sec,
+        ptype,
+        field,
+        fieldValues
+      }
+    };
+    // @ts-ignore - publish is exists.
+    await client.publish(this.channel, JSON.stringify(message));
+  }
+
+  public async updateForSavePolicy(model: Model): Promise<boolean> {
+    const client = await this.pubConnection.getRedisClient();
+    const message = {
+      updateType: updateTypes.savePolicy,
+      args: {
+        model
```

Comment:
There need to get the policy and grouping policy from the model, and then add them to `args`.